src/lobattocells.jl

```diff
@@ -318,6 +318,251 @@ function materializepoints(
     return points
 end
 
+function materializepoints(
+    referencecell::LobattoQuad,
+    coarsegridcells,
+    coarsegridvertices,
+    interpolation_degree,
+    faceinterpolation,
+    quadranttointerpolation,
+    quadranttolevel,
+    quadranttotreeid,
+    quadranttocoordinate,
+    forest,
+    comm,
+)
+    r = vec.(points_1d(referencecell))
+
+    FT = floattype(referencecell)
+    AT = arraytype(referencecell)
+    N = (interpolation_degree + 1, interpolation_degree + 1)
+    interp_r = vec.(points_1d(LobattoCell{Tuple{N...},FT,AT}()))
+
+    Q = max(512 ÷ prod(length.(r)), 1)
+
+    IntType = typeof(length(r))
+    num_local = IntType(P4estTypes.lengthoflocalquadrants(forest))
+    points = GridArray{eltype(coarsegridvertices)}(
+        undef,
+        arraytype(referencecell),
+        (length.(r)..., num_local),
+        (length.(r)..., length(quadranttolevel)),
+        comm,
+        false,
+        length(r) + 1,
+    )
+
+    backend = get_backend(points)
+
+    kernel! = curvedquadpoints!(backend, (length.(r)..., Q))
+    kernel!(
+        points,
+        r...,
+        interp_r...,
+        coarsegridcells,
+        coarsegridvertices,
+        interpolation_degree,
+        faceinterpolation,
+        quadranttointerpolation,
+        length(quadranttolevel),
+        quadranttolevel,
+        quadranttotreeid,
+        quadranttocoordinate,
+        Val.(length.(r))...,
+        Val(Q);
+        ndrange = size(points),
+    )
+
+    return points
+end
+
+@inline function interp(t, n, data, offset, degree)
+    x = zero(t)
+    y = zero(t)
+    t = 2 * t - 1
+    for i = 1:degree+1
+        li = one(t)
+        for j = 1:degree+1
+            if i != j
+                li *= (t - n[j]) / (n[i] - n[j])
+            end
+        end
+        x += data[offset+i-1, 1] * li
+        y += data[offset+i-1, 2] * li
+    end
+    return x, y
+end
+
+@inline function bary_interp(t, n, data, offset, degree)
+    x = zero(t)
+    y = zero(t)
+    denom = zero(t)
+    t = 2 * t - 1
+    w = ones(typeof(t), degree + 1)
+
+    for i in 1:degree+1
+        for j in 1:degree+1
+            if i ≠ j
+                w[i] *= (n[i] - n[j])
+            end
+        end
+        w[i] = 1.0 / w[i]
```

Comment:
```suggestion
        w[i] = 1 / w[i]
```

We don't want to use `Float64` literals because it will promote `Float32`s.